src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+  </video>
+  <figcaption>What we are going to build</figcaption>
+</figure>
+
+Let me start by saying: **supporting light and dark mode in an application or design system is hard.** Especially if your app is on multiple platforms and even more so if you have a large application. Light and dark mode on a personal site (like this one) is a whole different set of challenges. One more warning before moving on: 
+
+> Doing dark mode poorly is *worse* than not doing dark mode at all.
+
+Having certain pages or worse, certain components on a page, in light mode and others in dark mode is painful to look at. Even if you don't have that issue, making sure color contrast ratios meet accessibility standards and the visual hierarchies remain in both modes is a lot of work.
+
+Having said all that, using [design tokens](https://designtokens.org) can drastically simplify the creation and maintenance of color modes across platforms. I love that design tokens can integrate into any platform in a standard way by generating source code artifacts so that developers on a given platform don't even necessarily have to know they are interacting with design tokens. Building with design tokens on a given platform should feel natural. My approach to design token integration is to, as closely as possible, integrate with the built-in mechanisms in each platform to handle color modes rather than outputting some proprietary code.
+
+We are going to build a design token package using Style Dictionary that has support for light and dark mode for color tokens and graphic assets (using tokens as well). It will have demo apps in Android, iOS, and Web to show how it would integrate into those platforms. This article will serve as a high-level explanation of concepts; the why and how things work. The [Github repository](repo) has everything fully implemented with comments and more in-depth explanations. 
+
+This article and sample will not cover:
+* How to think about and build dark mode support from a UX perspective (like building a setting page or control for toggling light/dark mode)
+* Non-platform or framework-specific ways of implementing light/dark mode. For example, using a `<ThemeProvider>` in React or using a CSS-in-JS framework. You can absolutely still use those things if you want. I chose not to include every possible integration and instead stick to the native web, Android, and iOS mechanisms for light/dark mode.
+* React Native, Flutter, Android Jetpack Compose, Nativescript, and other cross-platform libraries because this article is already probably too long. But maybe I will cover those in a future article. 
+* A "complete" design token package or design system. To keep it short this will focus only on color tokens with different tiers, which will be explained later, and graphic assets as design tokens. I only built 2 very components: button, and badge with limited options just to illustrate how the color tokens work at a component level. 
```

Comment:
button, and badge with limited options just to illustrate > button and badge with limited options, just to illustrate